pkg/auth/csp/grpc.go

```diff
@@ -69,27 +69,30 @@ func (c *configSource) Token() (*oauth2.Token, error) {
 	if !g.IsGlobal() { // nolint:staticcheck // Deprecated
 		return nil, fmt.Errorf("trying to fetch token for non global server")
 	}
-	if !IsExpired(g.GlobalOpts.Auth.Expiration) {
-		tok := &oauth2.Token{
-			AccessToken: g.GlobalOpts.Auth.AccessToken,
-			Expiry:      g.GlobalOpts.Auth.Expiration,
+	var expiration time.Time
+	token := &Token{}
+	if g != nil && g.GlobalOpts != nil {
```

Comment:
Instead of adding `g != nil` here, it should be added at `L-66` and return an error.